### ISSUE TITLE
False positive for SC2026 when line-continuation backslash escape is immediately after the quotes

### ISSUE THREAD

**Issue by Lin-Buo-Ren** [koalaman/shellcheck#923]
- Rule Id: SC2026
- My shellcheck version: 0.4.6
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet or screenshot that shows the problem:

```bash
#!/usr/bin/env bash
declare \
        verbose='0'\
        debug='0'

echo "${verbose} ${debug}"
```

#### Here's what shellcheck currently says:
```
Line 3:
        verbose='0'\
                   ^-- SC2026: This word is outside of quotes. Did you intend to 'nest '"'single quotes'"' instead'? 
```

#### Here's what I wanted or expected to see:
```
No issues detected!
```

**Comment by Nightfirecat** [comment]
I don't think this is a false positive. Realistically, your code is functionally the same as writing the following, which generates no such notice. (note the added space)

```
#!/usr/bin/env bash
declare \
        verbose='0' \
        debug='0'

echo "${verbose} ${debug}"
```

For code style purposes, it makes more sense to me to add the extra space following the `verbose='0'` line, as there's no functional impact. Typically, placing backslashes immediately following a quoted string is used to continue long strings.

**Comment by Lin-Buo-Ren** [comment]
I do know that adding an extra space will inhibit the warning, however I disagree that to be in line with code styles and consider it as a workaround.

**Comment by Lin-Buo-Ren** [comment]
Also I believe it can be easily detected and ignored by ShellCheck, thus it should be fixed by so.

**Comment by brother** [comment]
Maybe it's possible to separate this particular warning from "others". It was designed to warn about incorrect single quotes and slashes and so on I believe and while you disagree with the warning I do think it does exactly what it was designed for in this case. Just as adding a directive to silence a warning for something else that you want adding a space is the way to silence this (or via directive ofc).

**Comment by koalaman** [comment]
While I would love to force everyone to add that space, this was not intentional :P 

The rule warns about single quoted strings followed by an all-alphanumeric literal. In this case, the escaped linefeed is a literal -- of length 0. All zero of them are of course alphanumeric. This has been fixed ce950ed.

**Comment by MarcSeebold** [comment]
In the [SC wiki for SC1004](https://github.com/koalaman/shellcheck/wiki/SC1004) the "correct code" does not have a white-space before the slash.

**Comment by brother** [comment]
> In the [SC wiki for SC1004](https://github.com/koalaman/shellcheck/wiki/SC1004) the "correct code" does not have a white-space before the slash. Actually, the "correct code" produces SC2026.

Are you using an older version or something?
I am not able to repeat it on [shellcheck.net](shellcheck.net) (no errors other than the warning about unused `$var`) for `bash` and `sh` shell types.
